src/VivViewer.js

```diff
@@ -1,157 +1,117 @@
 import React, { PureComponent } from 'react';
 import DeckGL from '@deck.gl/react';
-import { OrthographicView } from '@deck.gl/core';
-import { VivViewerLayer, StaticImageLayer } from './layers';
-import VivViewerOverview from './VivViewerOverview';
 
+/**
+ * This class handles rendering the various views within the DeckGL contenxt.
+ * @param {Array} layerProps The props for the layers in each view.
+ * @param {Array} viewStates The initial view states for each view.
+ * @param {VivView} views The various VivViews to render.
+ * */
 export default class VivViewer extends PureComponent {
   constructor(props) {
     super(props);
-    const {
-      initialViewState,
-      overview,
-      viewWidth,
-      viewHeight,
-      loader
-    } = this.props;
-    const vivOverview = overview
-      ? new VivViewerOverview({
-          ...overview,
-          viewWidth,
-          viewHeight,
-          loader,
-          margin: overview.margin,
-          overviewScale: overview.scale
-        })
-      : null;
     this.state = {
-      viewState: {
-        detail: { ...initialViewState, id: 'detail' },
-        overview: overview ? vivOverview.getViewState(initialViewState) : null
-      },
-      overview: vivOverview
+      viewStates: {}
     };
+    const { viewStates } = this.state;
+    const { views } = this.props;
+    views.forEach((view, i) => {
+      // eslint-disable-next-line react/destructuring-assignment
+      viewStates[view.id] = view.getViewState(this.props.viewStates[i]);
```

Comment:
What does the `getViewState` function do? How is `this.props.viewStates` different from `this.state.viewStates`? Does `getViewState` make some conversion?

Reply:
Yes, for example, the `OverviewView` holds various properties constant.  For example, you could see linked zoom-pan for side-by-side behaving differently according to flags for coupling and decoupling the zoom/pan.